src/main/java/reactor/netty/http/server/HttpServerOperations.java

```diff
@@ -261,12 +261,14 @@ public HttpMethod method() {
 
 	@Override
 	@Nullable
+	@SuppressWarnings("CollectionUndefinedEquality")
 	public String param(CharSequence key) {
 		Objects.requireNonNull(key, "key");
 		Map<String, String> params = null;
 		if (paramsResolver != null) {
 			params = this.paramsResolver.apply(uri());
 		}
+		// Suppressed "CollectionUndefinedEquality", the CharSequence is String
```

Comment:
the type is not enforced by the method signature though... what happens if at some point it becomes possible the calling code passes a different type of `CharSequence` like a `StringBuffer` or `StringBuilder` ?

I think there is an issue of brittle code here, and it would be preferable to pass `key.toString()` to the map get operation, which has 0 cost if the key is indeed a `String`.

(see also the advice on CharSequence [here](https://errorprone.info/bugpattern/UndefinedEquals))

Reply:
fixed